Recipe for using AVA with Webpack aliases

I use Webpack in my app and I defined aliases in config:
webpack.config.dev.js:

``` javascript
....
resolve: {
    // We can now require('file') instead of require('file.jsx')
    extensions: ['', '.js', '.jsx', '.scss'],
    alias: {
      actions: PATHS.app + '/actions',
      components: PATHS.app + '/components',
      store: PATHS.app + '/store',
      reducers: PATHS.app + '/reducers',
      helpers: PATHS.app + '/helpers',
      sagas:  PATHS.app + '/sagas',
      services:  PATHS.app + '/services',
    },
  }
....
```

And I import modules like so:
componts/LoginPage.js

```
import {login} from 'actions/Login'
```

instead of

```
import {login} from '../actions/Login'
```

But AVA doesn't allow to define aliases, so I'm unable to test **componts/LoginPage.js** with AVA since it throws that error:

```
Error: Cannot find module 'actions/Login'
```

Is there a solution?

--- a/docs/recipes/precompiling-with-webpack.md
+++ b/docs/recipes/precompiling-with-webpack.md
@@ -1,0 +1,25 @@
+In the [README](https://github.com/avajs/ava#transpiling-imported-modules) it mentions precompiling your imported modules as an alternative to JIT compilation, but it doesn't explain how. This recipe shows how to do this using webpack.
+
+webpack.config.js
+```js
+var nodeExternals = require('webpack-node-externals');
+
+module.exports = {
+  entry: ['./src/tests.js'],
+  target: 'node',
+  output: {
+    path: '_build',
+    filename: 'tests.js'
+  },
+  externals: nodeModules,
+  module: {
+    rules: [
+      { test: /\.(js|jsx)$/, use: 'babel-loader' }
+    ]
+  }
+};
+```
+
+The important bits are `target: 'node'`, which ignores node-specific `require`s (e.g. `fs`, `path`, etc.) and `externals: nodeModules` which prevents webpack from trying to bundle external node librarires which it may choke on.
+
+Now you can simply run `ava _buid/tests.js` to run the tests contained in this output.

--- a/docs/recipes/precompiling-with-webpack.md
+++ b/docs/recipes/precompiling-with-webpack.md
@@ -26,4 +26,4 @@ module.exports = {
 
 The important bits are `target: 'node'`, which ignores Node.js-specific `require`s (e.g. `fs`, `path`, etc.) and `externals: nodeModules` which prevents webpack from trying to bundle external Node.js libraries which it may choke on.
 
-You can now run `$ ava _buid/tests.js` to run the tests contained in this output.
+You can now run `$ ava _build/tests.js` to run the tests contained in this output.

--- a/docs/recipes/precompiling-with-webpack.md
+++ b/docs/recipes/precompiling-with-webpack.md
@@ -1,25 +1,29 @@
-In the [README](https://github.com/avajs/ava#transpiling-imported-modules) it mentions precompiling your imported modules as an alternative to JIT compilation, but it doesn't explain how. This recipe shows how to do this using webpack.
+## Precompiling source files with webpack
 
-webpack.config.js
+The AVA [readme](https://github.com/avajs/ava#transpiling-imported-modules) mentions precompiling your imported modules as an alternative to runtime compilation, but it doesn't explain how. This recipe shows how to do this using webpack.
+
+###### webpack.config.js
+
 ```js
-var nodeExternals = require('webpack-node-externals');
+const nodeExternals = require('webpack-node-externals');
 
 module.exports = {
-  entry: ['./src/tests.js'],
-  target: 'node',
-  output: {
-    path: '_build',
-    filename: 'tests.js'
-  },
-  externals: nodeExternals,
-  module: {
-    rules: [
-      { test: /\.(js|jsx)$/, use: 'babel-loader' }
-    ]
-  }
+	entry: ['src/tests.js'],
+	target: 'node',
+	output: {
+		path: '_build',
+		filename: 'tests.js'
+	},
+	externals: nodeExternals,
+	module: {
+		rules: [{
+				test: /\.(js|jsx)$/,
+				use: 'babel-loader'
+		}]
+	}
 };
 ```
 
-The important bits are `target: 'node'`, which ignores node-specific `require`s (e.g. `fs`, `path`, etc.) and `externals: nodeModules` which prevents webpack from trying to bundle external node librarires which it may choke on.
+The important bits are `target: 'node'`, which ignores Node.js-specific `require`s (e.g. `fs`, `path`, etc.) and `externals: nodeModules` which prevents webpack from trying to bundle external Node.js libraries which it may choke on.
 
-Now you can simply run `ava _buid/tests.js` to run the tests contained in this output.
+You can now run `$ ava _buid/tests.js` to run the tests contained in this output.

--- a/docs/recipes/precompiling-with-webpack.md
+++ b/docs/recipes/precompiling-with-webpack.md
@@ -1,6 +1,6 @@
 ## Precompiling source files with webpack
 
-The AVA [readme](https://github.com/avajs/ava#transpiling-imported-modules) mentions precompiling your imported modules as an alternative to runtime compilation, but it doesn't explain how. This recipe shows how to do this using webpack. (This example uses webpack 2.0.)
+The AVA [readme](https://github.com/avajs/ava#transpiling-imported-modules) mentions precompiling your imported modules as an alternative to runtime compilation, but it doesn't explain how. This recipe shows how to do this using webpack. (This example uses webpack 2.0)
 
 ###### webpack.config.js
 
@@ -24,6 +24,6 @@ module.exports = {
 };
 ```
 
-The important bits are `target: 'node'`, which ignores Node.js-specific `require`s (e.g. `fs`, `path`, etc.) and `externals: nodeModules` which prevents webpack from trying to bundle external Node.js libraries which it may choke on.
+The important bits are `target: 'node'`, which ignores Node.js-specific `require`s (e.g. `fs`, `path`, etc.) and `externals: nodeModules` which prevents webpack from trying to bundle external Node.js modules which it may choke on.
 
 You can now run `$ ava _build/tests.js` to run the tests contained in this output.

--- a/docs/recipes/precompiling-with-webpack.md
+++ b/docs/recipes/precompiling-with-webpack.md
@@ -1,6 +1,6 @@
 ## Precompiling source files with webpack
 
-The AVA [readme](https://github.com/avajs/ava#transpiling-imported-modules) mentions precompiling your imported modules as an alternative to runtime compilation, but it doesn't explain how. This recipe shows how to do this using webpack.
+The AVA [readme](https://github.com/avajs/ava#transpiling-imported-modules) mentions precompiling your imported modules as an alternative to runtime compilation, but it doesn't explain how. This recipe shows how to do this using webpack. (This example uses webpack 2.0.)
 
 ###### webpack.config.js
 

--- a/docs/recipes/precompiling-with-webpack.md
+++ b/docs/recipes/precompiling-with-webpack.md
@@ -11,7 +11,7 @@ module.exports = {
     path: '_build',
     filename: 'tests.js'
   },
-  externals: nodeModules,
+  externals: nodeExternals,
   module: {
     rules: [
       { test: /\.(js|jsx)$/, use: 'babel-loader' }

--- a/readme.md
+++ b/readme.md
@@ -734,7 +734,8 @@ AVA currently only transpiles the tests you ask it to run, as well as test helpe
 
 If you use Babel you can use its [require hook](https://babeljs.io/docs/usage/require/) to transpile imported modules on-the-fly. To add it, [configure it in your `package.json`](#configuration).
 
-You can also transpile your modules in a separate process and refer to the transpiled files rather than the sources from your tests. Example [here](docs/recipes/precompiling-with-webpack.md).
+You can also transpile your modules in a separate process and refer to the transpiled files rather than the sources from your tests. Example [here](docs/recipes/
+-with-webpack.md).
 
 ### Promise support
 
@@ -1120,7 +1121,7 @@ It's the [Andromeda galaxy](https://simple.wikipedia.org/wiki/Andromeda_galaxy).
 - [JSPM and SystemJS](docs/recipes/jspm-systemjs.md)
 - [Debugging tests with Chrome DevTools](docs/recipes/debugging-with-chrome-devtools.md)
 - [Debugging tests with WebStorm](docs/recipes/debugging-with-webstorm.md)
-- [Precompiling code with Webpack](docs/recipes/precompiling-with-webpack.md)
+- [Precompiling source files with webpack](docs/recipes/precompiling-with-webpack.md)
 
 ## Support
 

--- a/readme.md
+++ b/readme.md
@@ -734,7 +734,7 @@ AVA currently only transpiles the tests you ask it to run, as well as test helpe
 
 If you use Babel you can use its [require hook](https://babeljs.io/docs/usage/require/) to transpile imported modules on-the-fly. To add it, [configure it in your `package.json`](#configuration).
 
-You can also transpile your modules in a separate process and refer to the transpiled files rather than the sources from your tests.
+You can also transpile your modules in a separate process and refer to the transpiled files rather than the sources from your tests. Example [here](docs/recipes/precompiling-with-webpack.md).
 
 ### Promise support
 
@@ -1120,6 +1120,7 @@ It's the [Andromeda galaxy](https://simple.wikipedia.org/wiki/Andromeda_galaxy).
 - [JSPM and SystemJS](docs/recipes/jspm-systemjs.md)
 - [Debugging tests with Chrome DevTools](docs/recipes/debugging-with-chrome-devtools.md)
 - [Debugging tests with WebStorm](docs/recipes/debugging-with-webstorm.md)
+- [Precompiling code with Webpack](docs/recipes/precompiling-with-webpack.md)
 
 ## Support
 

--- a/readme.md
+++ b/readme.md
@@ -734,8 +734,7 @@ AVA currently only transpiles the tests you ask it to run, as well as test helpe
 
 If you use Babel you can use its [require hook](https://babeljs.io/docs/usage/require/) to transpile imported modules on-the-fly. To add it, [configure it in your `package.json`](#configuration).
 
-You can also transpile your modules in a separate process and refer to the transpiled files rather than the sources from your tests. Example [here](docs/recipes/
--with-webpack.md).
+You can also transpile your modules in a separate process and refer to the transpiled files rather than the sources from your tests. Example [here](docs/recipes/precompiling-with-webpack.md).
 
 ### Promise support
 

